pkg/route/validation/validation_test.go

```diff
@@ -843,6 +981,522 @@ func TestValidateRoute(t *testing.T) {
 	}
 }
 
+// This unit test tests if the Actions to set/delete headers were passed correctly in the Route Spec and checks if an appropriate error message was provided.
+func TestValidateHeaders(t *testing.T) {
+	var headerNameXFrame = "X-Frame-Options"
+	var headerNameXSS = "X-XSS-Protection"
+	var headerNameHSTS = "Strict-Transport-Security"
+	var headerNameProxy = "Proxy"
+	var headerNameCookie = "Cookie"
+	var headerNameSetCookie = "Set-Cookie"
+	tests := []struct {
+		name                 string
+		route                *routev1.Route
+		expectedErrorMessage string
+		headerType           string
+	}{
+		{
+			name: "Should give an error on attempt to delete the HSTS header.",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
```

Comment:
It would be easier to review and maintain if you left out the optional fields that are not required by this test.

Reply:
ok removed it. You remove from other tests also ?

Reply:
anyway that won't cause an issue and also won't affect our tests.